src/scripts/main.js

```diff
@@ -1,5 +1,40 @@
 'use strict';
 
-document.addEventListener('click', (e) => {
-  // write code here
+const wall = document.querySelector('.wall');
+const spider = wall.querySelector('.spider');
+
+document.addEventListener('click', (event) => {
+  const onWall = event.target.closest('.wall');
+
+  if (onWall) {
+    const x = event.clientX;
+    const y = event.clientY;
+    const styleWall = getComputedStyle(wall);
+    const rect = wall.getBoundingClientRect();
+    const borderWidth = parseFloat(styleWall.borderWidth);
+    const minLeft = rect.left + borderWidth + spider.offsetWidth / 2;
+    const minTop = rect.top + borderWidth + spider.offsetHeight / 2;
+    const maxLeft = rect.right - borderWidth - spider.offsetHeight / 2;
+    const maxTop = rect.bottom - borderWidth - spider.offsetHeight / 2;
+
+    if (x < minLeft) {
+      spider.style.left = `${0}px`;
+    } else {
+      spider.style.left = `${x - rect.left - spider.offsetWidth / 2 - parseFloat(styleWall.borderWidth)}px`;
+    }
+
+    if (y < minTop) {
+      spider.style.top = `${0}px`;
+    } else {
+      spider.style.top = `${y - rect.top - spider.offsetHeight / 2 - parseFloat(styleWall.borderWidth)}px`;
```

Comment:
The calculation for the spider's top position is correct, but ensure that the spider does not exceed the `maxTop` boundary. Consider adding a condition to check if `y - rect.top - spider.offsetHeight / 2 - parseFloat(styleWall.borderWidth)` exceeds `maxTop`.